lib/hopper.rb

```diff
@@ -33,6 +33,8 @@ def publish(message, key)
     end
 
     def add_subscriber(subscriber)
+      return if subscribers.include? subscriber
```

Comment:
How is the subscriber uniquely identified? Don't we need to override the equal? method on the subscriber class to identify a subscriber by the class+event+method tuple?